src/compiler/utilities.ts

```diff
@@ -8587,52 +8587,270 @@ export function getSetExternalModuleIndicator(options: CompilerOptions): (file:
     }
 }
 
-/** @internal */
-export function getEmitScriptTarget(compilerOptions: { module?: CompilerOptions["module"]; target?: CompilerOptions["target"]; }): ScriptTarget {
-    return compilerOptions.target ??
-        ((compilerOptions.module === ModuleKind.Node16 && ScriptTarget.ES2022) ||
-            (compilerOptions.module === ModuleKind.NodeNext && ScriptTarget.ESNext) ||
-            ScriptTarget.ES5);
+type CompilerOptionKeys = keyof { [K in keyof CompilerOptions as string extends K ? never : K]: any; };
+function createComputedCompilerOptions<T extends Record<string, CompilerOptionKeys[]>>(
+    options: {
+        [K in keyof T & CompilerOptionKeys]: {

```

Comment:
To get rid of the ` satisfies Record<StrictOptionName, any>` at the resulting object you could do this:

```suggestion
        [K in keyof T & CompilerOptionKeys | StrictOptionName]: {
```